pkg/controllers/gateway/gateway_controller_test.go

```diff
@@ -0,0 +1,821 @@
+									Listeners: []v1beta1.Listener{
+										{
+											Name:     testutil.ValidTestHostname,
+											Hostname: getTestHostname(testutil.ValidTestHostname),
+											Protocol: v1beta1.HTTPSProtocolType,
+										},
+									},
+								},
+							},
+						},
+					},
+					&v1beta1.GatewayClassList{
+						Items: []v1beta1.GatewayClass{
+							{
+								ObjectMeta: v1.ObjectMeta{
+									Name: testutil.DummyCRName,
+								},
+							},
+						},
+					},
+					getValidCertificateSecret(testutil.ValidTestHostname),
+					buildTestMZ(),
+					buildTestDNSRecord(),
+				),
+				Scheme: testutil.GetValidTestScheme(),
+			},
+			args: args{
+				req: testutil.BuildValidTestRequest(testutil.DummyCRName, testutil.Namespace),
+			},
+			verify: testutil.AssertNoErrorReconciliation(),
+		},
+		{
+			name: "failed to fetch gateway",
+			fields: fields{
+				Client: fake.NewClientBuilder().
+					WithScheme(runtime.NewScheme()).
+					Build(),
+				Scheme: runtime.NewScheme(),
+			},
+			args: args{
+				req: ctrl.Request{},
+			},
+			verify: func(res ctrl.Result, err error, t *testing.T) {
+				if !res.IsZero() || !strings.Contains(err.Error(), "no kind is registered") {
+					t.Errorf("failed. Err: %s, res: %v", err, res)
+				}
+			},
+		},
+		{
+			name: "no gateway is present",
+			fields: fields{
+				Client: testutil.GetValidTestClient(),
+				Scheme: testutil.GetValidTestScheme(),
+			},
+			args: args{
+				req: ctrl.Request{},
+			},
+			verify: testutil.AssertNoErrorReconciliation(),
+		},
+		{
+			name: "gateway is deleting",
+			fields: fields{
+				Client: testutil.GetValidTestClient(
+					&v1beta1.GatewayList{
+						Items: []v1beta1.Gateway{
+							{
+								ObjectMeta: v1.ObjectMeta{
+									Name:              testutil.DummyCRName,
+									Namespace:         testutil.Namespace,
+									DeletionTimestamp: testutil.GetTime(),
+								},
+							},
+						},
+					},
+				),
+				Scheme: testutil.GetValidTestScheme(),
+			},
+			args: args{
+				req: testutil.BuildValidTestRequest(testutil.DummyCRName, testutil.Namespace),
+			},
+			verify: testutil.AssertNoErrorReconciliation(),
+		},
+		{
+			name: "missing gateway class",
+			fields: fields{
+				Client: testutil.GetValidTestClient(
+					&v1beta1.GatewayList{
+						Items: []v1beta1.Gateway{
+							{
+								ObjectMeta: v1.ObjectMeta{
+									Name:       testutil.DummyCRName,
+									Namespace:  testutil.Namespace,
+									Finalizers: []string{GatewayFinalizer},
+								},
+								Spec: v1beta1.GatewaySpec{
+									GatewayClassName: testutil.DummyCRName,
+								},
+							},
+						},
+					},
+				),
+				Scheme: testutil.GetValidTestScheme(),
+			},
+			args: args{
+				req: testutil.BuildValidTestRequest(testutil.DummyCRName, testutil.Namespace),
+			},
+			verify: func(res ctrl.Result, err error, t *testing.T) {
+				if !reflect.DeepEqual(res, ctrl.Result{}) &&
+					!strings.Contains(err.Error(), "gatewayckasses") &&
```

Comment:
gatewayckasses?

Reply:
the funny part as it was passing with the typo. Fixed